lib/soapy_cake/client.rb

```diff
@@ -18,16 +18,24 @@ def run(request)
       request.api_key = api_key
       request.time_offset = time_offset
 
-      response = Response.new(http_response(request), request.short_response?)
-      response.time_offset = time_offset
-      response.to_enum
+      http_response(request).public_send(:"to_#{xml_response? ? 'xml' : 'enum'}")
```

Comment:
Why the magic? What's wrong with `xml_response? : response.to_xml : response.to_enum`?